replay/replay.go

```diff
@@ -232,8 +269,44 @@ func ReplayFile(t *testing.T, server pulumirpc.ResourceProviderServer, traceFile
 	assert.Greater(t, count, 0)
 }
 
+// See also: https://github.com/pulumi/pulumi/blob/master/pkg/util/rpcdebug/logformat.go#L28
 type jsonLogEntry struct {
 	Method   string          `json:"method"`
 	Request  json.RawMessage `json:"request,omitempty"`
 	Response json.RawMessage `json:"response,omitempty"`
+	Errors   []string        `json:"errors,omitempty"`
```

Comment:
https://github.com/pulumi/pulumi-terraform-bridge/pull/1496/ introduced Errors as *string but I think that's a problem because GRPC logs capture them as []string. So it inhibits grabbing GRPC logs and reusing them as asserts. Trying to fixup  here. Let me see that the tests from 1496 can still run with these changes.

Reply:
Yes it looks like with minor changes muxer_test can still work fine: https://github.com/pulumi/pulumi-terraform-bridge/pull/1638

Reply:
This PR contains an invalid go.mod. It is not demonstrative.

Reply:
I might have screwed up something with go workspaces. But I'm pretty sure the tests can be made to pass with some small changes. If not we can always iterate.

Reply:
I want to make sure that we can make tests pass before we merge. I'm comfortable merging with you saying "The tests have passed locally". We don't need to public facing CI here.

Reply:
I'm sure you overestimate the public interest in reading our PRs :)